File: e.VOC.us/Game/GameObjects/FakeShip.cs
using Newtonsoft.Json;

namespace e.VOC.us.Game.GameObjects
{
    public class FakeShip : GameObject
    {
        [JsonProperty("rectangle")]
        public RotatableRectangle Rectangle { get; set; }

        [JsonProperty("playerId")]
        public string PlayerId { get; set; }

        public FakeShip(RotatableRectangle rectangle, Player owner)
        {
            Rectangle = rectangle;
            PlayerId = owner.Id;
            TypeId = "fakeShip";
        }

        public override void Update(GameTime gametime)
        {
            
        }
    }
}

File: e.VOC.us/Scripts/TypeScript/FakeShip.ts
module eVOCus {
    export class FakeShip implements IServerObject {
        rectangle: RotatableRectangle;
        id: string;
        player: Player;
        animation: Animation;

        constructor(serverObj: any) {
            this.id = serverObj.id;
            this.rectangle = serverObj.rectangle;
            this.player = serverObj.player;
            var image = new Image();
            image.src = "/Assets/Boot-3.png";
            this.animation = new Animation(image, this.rectangle.width, this.rectangle.height * 5, 5, 2000, false);
        }

        update(gametime: number) {
            Game.instance.focus = this.rectangle.position;
        }

        synchronize(serverObj: any) {
        }

        draw(canvas: Canvas) {
            this.animation.Draw(canvas, 0, this.rectangle);
        }
    }
} 